File: backend/python/pdfToExcel.py
import sys
import os
import tabula
import pandas as pd

# Get the PDF file and output folder from the arguments
pdf_file = sys.argv[1]
output_folder = sys.argv[2]

# Set the output file path
output_file = os.path.join(output_folder, "output.xlsx")

# Ensure the output directory exists
os.makedirs(output_folder, exist_ok=True)

try:
    # Read tables from the PDF
    tables = tabula.read_pdf(pdf_file, pages='all', multiple_tables=True)

    # Convert the tables to a single DataFrame
    df = pd.concat(tables)

    # Save the DataFrame to an Excel file
    df.to_excel(output_file, index=False)

    print("success")
except Exception as e:
    print(f"error: {str(e)}")

File: backend/php/pdfToExcel.php
<?php
if ($_SERVER['REQUEST_METHOD'] == 'POST') {
    $pdfFile = $_FILES['file']['tmp_name'];

    // Define the output folder
    $outputFolder = '../../files/'; // Ensure this directory exists or is writable

    // Ensure the output folder exists
    if (!file_exists($outputFolder)) {
        mkdir($outputFolder, 0777, true);
    }
    // Validate the uploaded file is an Excel file
    $fileInfo = pathinfo($_FILES['file']['name']);
    $extension = strtolower($fileInfo['extension']);
    
    if ($extension !== 'pdf') {
        echo json_encode(['error' => 'Invalid file format. Please upload a Pdf Document.']);
        exit;
    }

    // Define the command to run the Python script
    $command = escapeshellcmd("python ../python/pdfToExcel.py " . escapeshellarg($pdfFile) . " " . escapeshellarg($outputFolder));
    $output = shell_exec($command);

    // Check if the command output indicates success
    if (trim($output) === "success") {
        // Return the converted file URL
        $outputFile = $outputFolder . $_FILES['file']['name'];
        echo json_encode(['success' => true, 'file_url' => '../../files/' . basename($outputFile)]);
    } else {
        // Handle errors
        echo json_encode(['error' => 'Pdf must be in a Table format']);
    }
}
